packages/hub/node-tests/services/contracts-test.ts

```diff
@@ -3,42 +3,48 @@ import { CONTRACT_EVENTS } from '../../services/contract-subscription-event-hand
 import { protocolVersions } from '@cardstack/cardpay-sdk';
+import semverGte from 'semver/functions/gte';
 
 describe('ContractsService', function () {
   let resultId = 1;
   for (const contractUsageInfo of CONTRACT_EVENTS) {
     for (const protocolVersion of protocolVersions) {
-      it(`getContract retrieves a web3 Contract instance for ${contractUsageInfo.abiName} contract version ${protocolVersion}`, async function () {
-        let subject = new ContractsService();
-        let web3 = new Web3();
-        let mockProvider = {
-          sendAsync(_, callback) {
-            callback(null, {
-              jsonrpc: '2.0',
-              id: resultId++,
-              result: web3.eth.abi.encodeParameter('string', protocolVersion.replace('v', '')),
-            } as any);
-          },
-        } as AbstractProvider;
-        web3.setProvider(mockProvider);
-        web3.eth.net.getId = (_callback) => {
-          return Promise.resolve(100);
-        };
-        let contract;
-        try {
-          contract = await subject.getContract(web3, contractUsageInfo.abiName, contractUsageInfo.contractName);
-        } catch (e) {
-          expect.fail(
-            `Failed to getContract for ${contractUsageInfo.abiName} contract.
-            
-  This may be due to a missing ABI definition. Check to see whether SDK codegen is working properly.
-            
-  ${e}`
-          );
-        }
-        expect(contract).to.be.an.instanceOf(web3.eth.Contract);
-        expect(contract).to.exist;
-      });
+      if (
+        contractUsageInfo.contractStartVersion &&
+        semverGte(protocolVersion, contractUsageInfo.contractStartVersion)
+      ) {
+        it(`getContract retrieves a web3 Contract instance for ${contractUsageInfo.abiName} contract version ${protocolVersion}`, async function () {
+          let subject = new ContractsService();
+          let web3 = new Web3();
+          let mockProvider = {
+            sendAsync(_, callback) {
+              callback(null, {
+                jsonrpc: '2.0',
+                id: resultId++,
+                result: web3.eth.abi.encodeParameter('string', protocolVersion.replace('v', '')),
+              } as any);
+            },
+          } as AbstractProvider;
+          web3.setProvider(mockProvider);
+          web3.eth.net.getId = (_callback) => {
+            return Promise.resolve(100);
+          };
+          let contract;
+          try {
+            contract = await subject.getContract(web3, contractUsageInfo.abiName, contractUsageInfo.contractName);
+          } catch (e) {
+            expect.fail(
+              `Failed to getContract for ${contractUsageInfo.abiName} contract.
+
+This may be due to a missing ABI definition. Check to see whether SDK codegen is working properly.
+
+${e}`
+            );
+          }
+          expect(contract).to.be.an.instanceOf(web3.eth.Contract);
+          expect(contract).to.exist;
+        });
+      }
```

Comment:
I'm not sure about this, but it's possible to append a skip clause so that skipped items have visibility:

```js
 else {
    it.skip(`Skipping ${contractUsageInfo.abiName} contract version ${protocolVersion} because earlier than start version ${contractUsageInfo.contractStartVersion}`, function () {});
}
```

The skipped items will be listed as pending tests.

Reply:
I’m going to skip (😆) this for now but we can revisit if it seems like it would have helped avoid a problem 🤓

Reply:
It would have helped avoid the comment above! I think